### ISSUE TITLE
How to use WriteInsideClassBody() ?

### ISSUE THREAD

**Issue by FaizulHussain** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#695]
I would like to write some boiler plate codes inside the generated entity/table class. I found this method "WriteInsideClassBody()" in the Table class. How/where do you overwrite this method in Database.tt? If I overwrite in "EF.Reverse.POCO.v3.ttinclude" then II'll end up losing it on subsequent upgrade / or is not flexible like other customization delegates.

Also, accessing NameHumanCase property inside WriteInsideClassBody() seems to return inconsistent value. In some cases the class names are plural but is singular inside the method, or vice versa. I think its the order of name sanitization causing this. Will it be possible to place the boiler plate codes at the end of the properties? At the moment it appears right at the top of all properties. I am trying to write some boiler plate methods :)

Just found that some of the column names (NameHumanCase) is also in consistent inside WriteInsideClassBody() method calls

Many Thanks

**Comment by sjh37** [comment]
Hi @FaizulHussain Yes you are correct. I need to make a change to expose it via `<database>.tt` instead of being inside `EF.Reverse.POCO.v3.ttinclude` which can change.

I checked `NameHumanCase` by using
```c#
public string WriteInsideClassBody()
{
    return "    // " + NameHumanCase + Environment.NewLine;
}
```
and they all matched the name of the class correctly. My database is not the same as yours so I am not seeing it. Could you send me some SQL to create a few tables where you see a mix of plural/singular mix up in `WriteInsideClassBody()`

**Comment by FaizulHussain** [comment]
I was trying to generate default ASP.NET identiy models. I think the issue is around the configs:
Settings.TableSuffix = "Entity";
Settings.TableRename = delegate(string name, string schema, bool isView)
    {
        return name + (isView ? "View" : "Entity");
    };

I'll try replicating it later this week (I've moved on to something else)

Thanks

**Comment by sjh37** [comment]
You can now use WriteInsideClassBody in your settings. Grab the latest [EF.Reverse.POCO.v3.ttinclude](https://raw.githubusercontent.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/7d9275fd9c3eab0dc6ea217ad39d0ad50c94aa75/EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.v3.ttinclude) file and add this to your `<database>.tt` file:
```cs
// Writes any boilerplate stuff inside the POCO class body
Settings.WriteInsideClassBody = delegate(Table t)
{
    // Example:
    //return "    // " + t.NameHumanCase + Environment.NewLine;

    // Do nothing by default
    return string.Empty;
};
```

This will be in the next release.

**Comment by sjh37** [comment]
p.s. Do not try and generate the ASP.NET identity models as they are already built in to the EF library. Hence the default use of the filter
```cs
FilterSettings.TableFilters.Add(new RegexExcludeFilter("AspNet.*")); // This excludes all tables starting with 'AspNet'
```

**Comment by sjh37** [comment]
Released in v3.4.1
